src/main/java/org/opencds/cqf/providers/FHIREndpointProvider.java

```diff
@@ -0,0 +1,149 @@
+package org.opencds.cqf.providers;
+
+import ca.uhn.fhir.jpa.dao.IFhirSystemDao;
+import ca.uhn.fhir.jpa.rp.dstu3.EndpointResourceProvider;
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Operation;
+import ca.uhn.fhir.rest.annotation.OptionalParam;
+import ca.uhn.fhir.rest.annotation.RequiredParam;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.client.api.ServerValidationModeEnum;
+import ca.uhn.fhir.rest.client.interceptor.BasicAuthInterceptor;
+import ca.uhn.fhir.rest.param.StringAndListParam;
+import ca.uhn.fhir.rest.param.StringOrListParam;
+import ca.uhn.fhir.rest.param.StringParam;
+import org.hl7.fhir.dstu3.model.*;
+import org.opencds.cqf.exceptions.Helper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FHIREndpointProvider extends EndpointResourceProvider {
+
+    private JpaDataProvider provider;
+    private IFhirSystemDao systemDao;
+
+    public FHIREndpointProvider(JpaDataProvider provider, IFhirSystemDao systemDao) {
+        this.provider = provider;
+        this.systemDao = systemDao;
+    }
+
+    @Operation(name="cache-valuesets", idempotent = true)
+    public Resource cacheValuesets(
```

Comment:
I expected the "cache-valuesets" operation to be on a library. Where do we get the valuesets to cache in this operation?

Reply:
I was under the impression that the library op would be Library/$cache-terminology, which would use this operation. This operation is the ValueSet/$cache operation we discussed.